### ISSUE TITLE
[a11y]: File uploader status indicators need programmatic equivalent

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11200]
### Package

@carbon/react

### Browser

Chrome

### Package version

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader

### Automated testing tool and ruleset

n/a

### Description

After a user selects a file to upload from their operating system, the status of the upload is revealed through a series of images which provide no text equivalent.

 To meet [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) and [1.1.1 Non-text content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- the 2 states of the file upload mechanism should be given an equivalent status message
<img width="269" alt="image" src="https://media.github.ibm.com/user/1900/files/b75c3480-b41c-11ec-8af7-4b20ffe9b063">
status message "uploading"

---

<img width="266" alt="Screenshot 2022-04-04 at 1 38 38 PM" src="https://media.github.ibm.com/user/1900/files/ccd15e80-b41c-11ec-87d2-91d04d29dd2b">
Status message "uploaded" OR possibly "[file name] uploaded"

### Solution

Provide a text equivalent, either by making the images visible to assistive technology and providing them with ALT text OR by putting them into a live region and providing text updates to that region.
You could also potentially use a similiar technique to [#11198 ](https://github.com/carbon-design-system/carbon/issues/11198), using the filename as an ariadescribedby on the icon, so that the user heard 'uploaded, [filename]'

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader

### Steps to reproduce

1. Visit https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader
2. use the Add file button to choose a file from your system. Choose I file close to the size limit (i.e., just under 500kb).
3. The file name will appear on the page, with an icon for 'uploading' and 'uploaded' to the right (before it becomes the 'delete')

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
@joshblack I think the information I've gathered in the [Inline loading topic](https://ibm.box.com/s/i7d5762pyf40idov5tpzbj4b7iv6td7w) is relative to this.

**Comment by mbgower** [comment]
There is also some question about focus handling here, both while adding a file and when deleting a file that has been added for uploading.
- Focus could go to X of file added, (and X of preceding file, when deleted)
- Focus could go to the Add file button
- Focus could go to the file name (and focus doesn't HAVE to necessarily be visible)
There is also the question about conveying the status of the upload/delete, and the necessary/best way to surface it programmatically.
